rust/agents/relayer/src/msg/pending_message.rs

```diff
@@ -414,10 +458,21 @@ impl PendingMessage {
         pm
     }
 
-    fn on_reprepare(&mut self) -> PendingOperationResult {
+    fn on_reprepare<E: Debug>(
+        &mut self,
+        err: E,
+        reason: ReprepareReason,
+    ) -> PendingOperationResult {
         self.inc_attempts();
         self.submitted = false;
-        PendingOperationResult::Reprepare
+        warn!(error = ?err, "{}", reason.clone());
+        PendingOperationResult::Reprepare(reason)
+    }
+
+    fn on_reconfirm<E: Debug>(&mut self, err: E, reason: &str) -> PendingOperationResult {
+        self.inc_attempts();
+        warn!(error = ?err, "{}", reason.clone());
```

Comment:
Wonder if adding a string in here to make this log line easy to search for may be useful, e.g. `"Reconfirming message: {}"` or something - same for the on_reprepare